Add article for JavaScript `typeof` keyword

david-daly opened the pull request:
Add some basic information about the JavaScipt `typeof` keyword, why it is useful and link to MDN Documentation fro more information.

13thThief commented:
Hello @david-daly 
There is a spelling mistake and this line, "The typeof operator is important in JavaScript as it allows programmers to easily check the type of a variable in their code. This is important as JavaScript is a language that does not require variables to have types assigned when they are created (it is a dynamically typed language)." adds complexity as the word 'important' is repeated twice.

david-daly commented:
Thanks for the feedback @13thThief !

I have fixed the phrasing in the sentence you mentioned, and fixed any spelling mistakes that I had in the first version.

HKuz commented:
LGTM! Thanks for another contribution, @david-daly ! 🥇 🌮 🏆
